packages/rpc-core/src/rpc-methods/__tests__/get-genesis-hash-test.ts

```diff
@@ -9,6 +9,24 @@ import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
 const logFilePath = path.resolve(__dirname, '../../../../../test-ledger/validator.log');
 const genesisHashPattern = /genesis hash: ([\d\w]{32,})/;
 
+async function getGenesisHashFromLogFile() {
+    return new Promise<string | undefined>((resolve, reject) => {
+        const logFileStream = fs.createReadStream(logFilePath);
+        logFileStream.on('error', e => reject(e))
+
+        logFileStream.on('data', data => {
+            const chunk = data.toString('utf-8');
+            const expectedGenesisHash = chunk.match(genesisHashPattern)?.[1];
+            if (expectedGenesisHash) {
+                logFileStream.destroy();
+                resolve(expectedGenesisHash)
+            }
+        })
+
+        logFileStream.on('end', () => resolve(undefined));
```

Comment:
This should be good, just curious if you think we should have some kind of retry loop, like `numRetries = 2`

Reply:
Do you mean we should just fail if it's not in the first N chunks? I think that if we want to do that the best way might be to set the `end` parameter when creating the stream, so it only reads X bytes (https://nodejs.org/api/fs.html#filehandlecreatereadstreamoptions). We don't seem to have much control over how big chunks are (I think `highWaterMark` sets an upper bound, that seems to be it) so I'm a bit hesitant to rely on a chunk count. But maybe we could just set the `end` at about 16Kb or something?

Reply:
No actually I was thinking what if it doesn't find the genesis hash in it's first attempt to read the chunks (maybe the file is only a few lines and the validator is still spinning up)?

The more I think about it, the more I feel like it's not an issue since if the validator is not in a ready state, everything's going to fail anyway.

If this causes any issues with any tests across the board, we're better off making an issue and adjusting [this bad boy](https://github.com/solana-labs/solana-web3.js/blob/master/scripts/start-shared-test-validator.sh)

Reply:
Yep agreed, the existing code seems to work fine in CI too just reading it right away. I've added a 64kb `end` just in case someone somehow ends up with a very broken validator.log. It's about 4.5kb into the file for my current one so pretty sure it'll always end up in the first 64kb!

Reply:
Ship it!

Reply:
Even easier, tho :)

https://github.com/solana-labs/solana-web3.js/pull/1478#discussion_r1300523142